spec/timezone.html

```diff
@@ -657,11 +647,26 @@ <h1>
         <dd>
           This operation is the internal implementation of the `Temporal.TimeZone.prototype.getOffsetStringFor` method.
           If the given _timeZone_ is an Object, it observably calls _timeZone_'s `getOffsetNanosecondsFor` method.
+          If the offset represents an integer number of minutes, then the output will be formatted like ±HH:MM.
+          Otherwise, the output will be formatted like ±HH:MM:SS or (if the offset does not evenly divide into seconds) ±HH:MM:SS.fff… where the "fff" part is a sequence of at least 1 and at most 9 fractional seconds digits with no trailing zeroes.
         </dd>
       </dl>
       <emu-alg>
         1. Let _offsetNanoseconds_ be ? GetOffsetNanosecondsFor(_timeZone_, _instant_).
-        1. Return FormatOffsetTimeZoneIdentifier(_offsetNanoseconds_, ~separated~).
+        1. Let _offsetMinutes_ be truncate(_offsetNanoseconds_ / (6 × 10<sup>10</sup>)).
+        1. Let _offsetString_ be FormatOffsetTimeZoneIdentifier(_offsetMinutes_ × (6 × 10<sup>10</sup>), ~separate~).
+        1. Let _subMinuteNanoseconds_ be abs(_offsetNanoseconds_) modulo (6 × 10<sup>10</sup>).
```

Comment:
I can't help but notice the smell of FormatOffsetTimeZoneIdentifier and GetOffsetStringFor performing some of the same arithmetic, and for that matter overlapping so much with FormatSecondsStringPart. It would be great to see reduction of FormatOffsetTimeZoneIdentifier to something like `Assert: _offsetNanoseconds_ modulo (6 × 10<sup>10</sup>) = 0. Return FormatUTCOffset(_offsetNanoseconds_, *":"*).` and GetOffsetStringFor to something like `Let _offsetNanoseconds_ be ? GetOffsetNanosecondsFor(_timeZone_, _instant_). Return FormatUTCOffset(_offsetNanoseconds_, *":"*).`, with FormatUTCOffset calculating _sign_ and returning the concatenation of it with the result of sending the decomposition of nanoseconds to a time-formatting operation (which should contain **all** of the hh:mm[:ss[.fff…]] formatting and be parameterized by both precision and colon-vs.-empty-string separator).

That said, I can understand separating such work out from this PR. Maybe it should precede it, though?

Reply:
I agree, it's a good idea for those three AOs to be coalesced editorially into something cleaner. There's also a \*lot\* of normative PRs waiting to get merged, and it's been pretty hard to coordinate all the PRs and Test262 changes.

So instead of adding another editorial roadblock before merging this one, I'd prefer to merge this PR first. I'll volunteer to build an editorial PR to simplify hh:mm[:ss[.fff…]]  formatting after all the offset-related PR dust has settled.

Sound OK?

Reply:
Yep, works for me.